src/PKSim.Core/Snapshots/Mappers/RandomPopulationSettingsMapper.cs

```diff
@@ -61,9 +84,43 @@ public override async Task<RandomPopulationSettings> MapToModel(PopulationSettin
          await updateModelRange(settings, CoreConstants.Parameters.MEAN_HEIGHT, snapshot.Height);
          await updateModelRange(settings, Constants.Parameters.GESTATIONAL_AGE, snapshot.GestationalAge);
          await updateModelRange(settings, CoreConstants.Parameters.BMI, snapshot.BMI);
+
+         await addDiseaseStateParameters(settings, snapshot);
          return settings;
       }
 
+      private async Task addDiseaseStateParameters(RandomPopulationSettings randomPopulationSettings, PopulationSettings snapshot)
+      {
+         if (snapshot.DiseaseStateParameters == null)
+            return;
+
+         //Can only be null if someone messes around with snapshots or in tests
+         var diseaseState = randomPopulationSettings.BaseIndividual?.OriginData?.DiseaseState;
+         if (diseaseState == null)
+            return;
+
+         foreach (var snapshotDiseaseStateParameter in snapshot.DiseaseStateParameters)
+         {
+            var diseaseStateParameter = diseaseState.Parameter(snapshotDiseaseStateParameter.Name);
+            if (diseaseStateParameter == null)
+               continue;
```

Comment:
shouldn't we show a warning here? "Snapshot disiease state parameter XY not found in ..."

Reply:
Ok

Reply:
Done